test/Algebra/Graph/Test/RewriteRules.hs

```diff
@@ -102,3 +102,15 @@ fmapFmap1 g f h = fmap h (fmap f g)
 fmapFmapR g f h = fmap (h . f) g
 
 inspect $ 'fmapFmap1 === 'fmapFmapR
+
+bind2, bind2R :: (a -> Graph b) -> (b -> Graph c) -> Graph a -> Graph c
+bind2 f g x = x >>= f >>= g
+bind2R f g x = x >>= (\x -> f x >>= g)
+
+inspect $ 'bind2 === 'bind2R
+
+ovSeqAp, seqApOv :: Graph (a -> b) -> Graph (a -> b) -> Graph a -> Graph b
+ovSeqAp x y z = (overlay x y) <*> z
+seqApOv x y z = overlay (x >>= (<$> z)) (y >>= (<$> z))
```

Comment:
Why can't we make proper left distributivity work `overlay x y <*> z == overlay (x <*> z) (y <*> z)`? 

What about the right distributivity `x <*> overlay y z == overlay (x <*> y) (x <*> z)`?

Reply:
Oops, sorry for the lack of explanation, I forget to write them !

So I am hitting a little problem:  if I use directly
```Haskell
seqApOv x y z = overlay (x <*> z) (y <*> z)
```
`<*>` is only replaced by `seqApR` and `seqApR` is then not inlined, and I don't know why...
The only way I found to force its inlining is adding the corresponding rewrite rule, but this is not very satisfying.
Note that `<*>` is correctly inlined in `ovSeqAp`.

So I just inlined it "by hand" but this is not very beautiful ^^.

I think the problem comes from the fact that `<*>` is a class function, but I don't really know how to solve this...

Reply:
In this case, it may be better to actually document the problem, and create an inspection test for the inequality (e.g. `=/=`). At some point this will hopefully be resolved and we'll swap to `===`.

Reply:
This is corrected, I added both tests with an explanation.

Sadly, I am not able to make right-distributivity works right now, and in fact I don't know if it is true (from a structural point of view)

Reply:
Thanks! Let me think about right distributivity for a day or two... I'm curious whether it actually works or not :)

Reply:
Ah indeed, you are right :)